src/NuGet.Core/NuGet.Protocol.Core.v3/Utility/GetDownloadResultUtility.cs

```diff
@@ -51,12 +57,25 @@ public static class GetDownloadResultUtility
                                 return new DownloadResourceResult(DownloadResourceResultStatus.NotFound);
                             }
 
-                            return await GlobalPackagesFolderUtility.AddPackageAsync(
-                                identity,
-                                packageStream,
-                                settings,
-                                logger,
-                                token);
+                            if (cacheContext.NoCache)
+                            {
+                                return await AddPackageDirectAsync(
+                                    identity,
+                                    packageStream,
+                                    settings,
+                                    cacheContext.GeneratedTempFolder,
+                                    logger,
+                                    token);
+                            }
+                            else
+                            {
+                                return await GlobalPackagesFolderUtility.AddPackageAsync(
```

Comment:
I would prefer a conditional -- do you think it's reasonable in this case? I'm not set on it but I try my very hardest to dedupe logically similar code.